bin/steps/sqlite3

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+set +e
+
+# shellcheck source=bin/utils
+source $BIN_DIR/utils
+
+SQLITE3_VERSION=$(apt-cache policy libsqlite3-dev | grep Candidate | sed 's/  Candidate: //')
+
+SQLITE3_VERSION_FILE="$BUILD_DIR/.heroku/python-sqlite3-version"
+if [ -f $SQLITE3_VERSION_FILE ]; then
+  INSTALLED_SQLITE3_VERSION=$(cat $SQLITE3_VERSION_FILE)
+fi
+
+# only install if the sqlite3 version has changed
+if [[ $INSTALLED_SQLITE3_VERSION != $SQLITE3_VERSION ]]; then
+  APT_CACHE_DIR="$CACHE_DIR/.heroku/python/apt/cache"
+  APT_STATE_DIR="$CACHE_DIR/.heroku/python/apt/state"
+
+  mkdir -p "$APT_CACHE_DIR/archives/partial"
+  mkdir -p "$APT_STATE_DIR/lists/partial"
+
+  APT_OPTIONS="-o debug::nolocking=true"
+  APT_OPTIONS="$APT_OPTIONS -o dir::cache=$APT_CACHE_DIR"
+  APT_OPTIONS="$APT_OPTIONS -o dir::state=$APT_STATE_DIR"
+  APT_OPTIONS="$APT_OPTIONS -o dir::etc::sourcelist=/etc/apt/sources.list"
+
+  puts-step "Installing SQLite3"
+  apt-get $APT_OPTIONS update >/dev/null 2>&1
+  apt-get $APT_OPTIONS -y -d --reinstall install libsqlite3-dev sqlite3 >/dev/null 2>&1
+
+  for deb in $(find $APT_CACHE_DIR/archives/ -name "*.deb"); do
+    dpkg -x $deb "$BUILD_DIR/.heroku/python/sqlite3/"
+  done
+
+  HEROKU_PYTHON_DIR="$BUILD_DIR/.heroku/python"
```

Comment:
The Python directory is referenced manually a few times above - perhaps it would be worth moving `HEROKU_PYTHON_DIR` up higher and using for those instances too?